### PR TITLE
Find-merge hell cleanup: TextBlockMatch simplification

### PR DESCRIPTION
**Pull request by jotare**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb/src/nucliadb/common/external_index_providers/pinecone.py [hunk 1]
```diff
@@ -84,23 +85,22 @@ def iter_matching_text_blocks(self) -> Iterator[TextBlockMatch]:
...
-                position_end_seconds=list(map(int, vector_metadata.position_end_seconds or [])),
                 is_a_table=vector_metadata.is_a_table or False,
                 page_with_visual=vector_metadata.page_with_visual or False,
                 representation_file=vector_metadata.representation_file,
                 paragraph_labels=vector_metadata.paragraph_labels or [],
                 field_labels=vector_metadata.field_labels or [],
-                page_number=vector_metadata.page_number,
+                position=TextPosition(
+                    page_number=vector_metadata.page_number,
+                    index=vector_id.index or 0,
```
Comment (javitonino) [inline nucliadb/src/nucliadb/common/external_index_providers/pinecone.py:98]: Before it didin't have the `or 0`. Why is it being added? I guess it's a None, default to 0 now, but is that correct?

File: nucliadb/src/nucliadb/common/external_index_providers/base.py [hunk 2]
```diff
@@ -59,16 +61,8 @@ class TextBlockMatch(BaseModel):
...
-    resource_id: str
-    field_id: str
-    subfield_id: Optional[str] = None
-    index: Optional[int] = None
-    position_start: int
-    position_end: int
-    position_start_seconds: list[int] = []
-    position_end_seconds: list[int] = []
-    page_number: Optional[int] = None
+    id: ParagraphId
```
Comment (javitonino) [inline nucliadb/src/nucliadb/common/external_index_providers/base.py:64]: Maybe rename this as `paragraph_id` or just `paragraph`? As it is now, looks like the id of the match, whatever that may mean. It confused me a little.

**Review by javitonino — APPROVED** [review]
This is correct as far as I can see, should be functionally equivalent. Left a question and a suggestion.

File: nucliadb/src/nucliadb/common/external_index_providers/pinecone.py (hunk 1 above)
Comment (jotare) [reply nucliadb/src/nucliadb/common/external_index_providers/pinecone.py:98]: It didn't have it there, but the only place where it was used was in the hydrate, where it computed the `TextPosition` and set `index=text_block.index or 0`, so I moved it here

**Review by lferran — APPROVED** [review]

**Comment by jotare** [comment]
Up to this point, I've completely transitioned from `TempFindParagraph` to `TextBlockMatch`. 

This is a needed step towards hydrator unification between our index node retrieval and external indexes. Remaining changes include:
- use the hydrator module in find_merge.py
- move multi match booster reranker logic outside find_merge.py
- proper reranker usage (without need to fetch unused resource metadata)

File: nucliadb/src/nucliadb/search/search/find_merge.py [hunk 3]
```diff
@@ -325,47 +263,58 @@ def rank_fusion_merge(
...
+                    page_number=vector.metadata.position.page_number,
+                    index=vector.metadata.position.index,
+                    start=vector_id.vector_start,
+                    end=vector_id.vector_end,
+                    start_seconds=[x for x in vector.metadata.position.start_seconds],
+                    end_seconds=[x for x in vector.metadata.position.end_seconds],
+                ),
+                # TODO: get labels from index
+                field_labels=[],
+                paragraph_labels=[],
```
Comment (lferran) [inline nucliadb/src/nucliadb/search/search/find_merge.py:313]: is it a problem if this is not populated here?

Comment (jotare) [reply nucliadb/src/nucliadb/search/search/find_merge.py:313]: It's not populated anywhere, but it wasn't either. There was a TODO saying "get labels from index"

**Pull request merged**